### PR TITLE
Grid upgrade

### PR DESCRIPTION
**Pull request by lbwexler**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by lbwexler** [comment]
Not quite sure how to handle this.  It looks like the type for headerName: is 'string' although the ReactNode form that we use in our column value filter implementation continues to work fine.

Given that the filter is an implementation class and continues to work well, I see no reason for immediate change, but we should probably respect agGrid's specified string type in our Column and ColumnGroup class.

**Comment by lbwexler** [comment]
Not sure if needs to be called out as a breaking change.  we could canvas people to see if it is being used to return ReactNodes elsewhere

**Comment by TomTirapani** [comment]
I don't understand the need to change the `headerName` type? It doesn't look like this is ever used directly by agGrid - the agGrid column uses it in its `headerValueGetter`, which already includes an `isString` check. Display is handled by our custom ColumnHeader component.

If there's no actual issue with using ReactNodes, why can't we keep the current types and avoid the breaking change? I know of a few cases where we use icons for header names.

**Comment by lbwexler** [comment]
Ah, I think I understand this better.  The problem is really just for *ColumnGroup*, where we don't have a displayName to fall back on if the user is supplying a react Node for the header name.

File: cmp/grid/columns/ColumnGroup.ts [hunk 1]
```diff
@@ -103,10 +103,15 @@ export class ColumnGroup {
...
-                isFunction(headerName)
+            headerValueGetter: agParams => {
+                // headerValueGetter should always return a string
+                // for display in draggable shadow box, aGrid Tool panel.
+                // Hoist ColumnHeader will handle display of Element values in the header.
+                const ret = isFunction(headerName)
                     ? headerName({columnGroup: this, gridModel, agParams})
-                    : headerName,
+                    : headerName;
+                return isString(ret) ? ret : genDisplayName(this.groupId);
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply cmp/grid/columns/ColumnGroup.ts:113]: we actually throw right above to protect ourselves from this.

**Pull request merged**